web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -33,17 +39,81 @@ export async function sendAndConfirmTransaction(
     sendOptions,
   );
 
-  const status = (
-    await connection.confirmTransaction(
-      signature,
-      options && options.commitment,
-    )
```

Comment:
Yes, that's the goal. I saw your comment about teaching `confirmTransaction` to use `latesetBlockhash` so that was the direction we wanted to take. I'll get this updated to reflect that change.

Reply:
Okay, I went ahead and started working on this. I'm still not used to working with overload functions in TypeScript, so I did run into some issues. From what I could find while researching this, TypeScript only allows function overloading if the number of arguments are the same, which means I couldn't use two separate signatures that each had a different number of arguments. Instead, I added a new method `confirmTransactionBlockHeight` that uses the new approach and deprecated `confirmTransaction`. Then, in `sendAndConfirmTransaction` I added a variable to hold `recentBlockhash` and `lastValidBlockHeight` to use with the new method. I was looking at `sendTransaction` and I noticed that `_blockhashInfo` only uses `recentBlockhash`. Do you think we should update these as well?

Reply:
I managed to figure this out. I had to disable the `no-dupe-class-members` eslint rule, and I wasn't getting issues. I tend not to mess with eslint rules when it's not my codebase, so if there's a reason this rule shouldn't be disabled I left the other implementation in (commented out). They're both there for comparison, so let me know what you think is best.